
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>RnaSeq JIP Pipeline &mdash; RNA Seq JIP Pipeline 0.0 documentation</title>
    
    <link rel="stylesheet" href="_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="RNA Seq JIP Pipeline 0.0 documentation" href="#" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li><a href="#">RNA Seq JIP Pipeline 0.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="rnaseq-jip-pipeline">
<h1>RnaSeq JIP Pipeline<a class="headerlink" href="#rnaseq-jip-pipeline" title="Permalink to this headline">¶</a></h1>
<p>Collection of <a class="reference external" href="https://pyjip.readthedocs.org/en/latest/">JIP pipeline system</a> scripts to manage the RNA Seq Pipeline using <a class="reference external" href="http://gemtools.github.io/docs/index.html">gemtools rnaseq</a> for mapping, <a class="reference external" href="http://sammeth.net/confluence/display/FLUX/Home">flux</a> for quantification and <a class="reference external" href="http://lomereiter.github.io/sambamba/">sambamba</a> to calculate PCR Duplicates statistics. It also updates a temporary file to be processed by <a class="reference external" href="http://statgen.cnag.cat/daemon_rna_lims/">daemon lims</a> in order to update the pipeline results to the Lims System.</p>
<div class="section" id="prerequisites">
<h2>0. Prerequisites<a class="headerlink" href="#prerequisites" title="Permalink to this headline">¶</a></h2>
<div class="section" id="download-and-install-jip-pipeline-system">
<h3>0.1. Download and install JIP PIPELINE system<a class="headerlink" href="#download-and-install-jip-pipeline-system" title="Permalink to this headline">¶</a></h3>
<p>The JIP pipeline system is a python library and a set of command line utilities that allows you to create batch-process based computational pipeline that can be submitted and managed on a computer cluster or on your local machine.</p>
<p>Install the jip pipeline system following the instructions found in: <a class="reference external" href="http://pyjip.readthedocs.org/en/latest/">http://pyjip.readthedocs.org/en/latest/</a></p>
</div>
<div class="section" id="download-and-install-gemtools-rnaseq">
<h3>0.2. Download and install gemtools rnaseq<a class="headerlink" href="#download-and-install-gemtools-rnaseq" title="Permalink to this headline">¶</a></h3>
<p>Get gemtools rnaseq from <a class="reference external" href="http://gemtools.github.io/">http://gemtools.github.io/</a> and follow install instructions.</p>
</div>
<div class="section" id="flux-capacitor">
<h3>0.3 Flux-Capacitor<a class="headerlink" href="#flux-capacitor" title="Permalink to this headline">¶</a></h3>
<p>Considers the read abundance on an exonic segment as the cumulative abundance of all transcript isoforms. The transcript is represented as a combination of exons and aims at estimating individual transcript abundance from the observed read counts at each exon.</p>
<p>Download and Install flux capacitor from <a class="reference external" href="http://sammeth.net/confluence/display/FLUX/2+-+Download">http://sammeth.net/confluence/display/FLUX/2+-+Download</a></p>
</div>
<div class="section" id="download-and-install-sambamba">
<h3>0.4. Download and install SAMBAMBA<a class="headerlink" href="#download-and-install-sambamba" title="Permalink to this headline">¶</a></h3>
<p>Sambamba is a tool for processing SAM/BAM files in a fast way. It is used for merging, calculate PCR Duplicates and general statistics from BAM files.</p>
<p>Download and install the package in your system from: <a class="reference external" href="https://github.com/lomereiter/sambamba/releases">https://github.com/lomereiter/sambamba/releases</a></p>
</div>
<div class="section" id="lims-updating-daemon">
<h3>0.5. Lims updating daemon<a class="headerlink" href="#lims-updating-daemon" title="Permalink to this headline">¶</a></h3>
<p>If you want to use the lims updating system to upload the data generated by the pipeline you must install and start the <a class="reference external" href="http://statgen.cnag.cat/daemon_rna_lims/">rna updating lims daemon</a>.</p>
</div>
</div>
<div class="section" id="install">
<h2>1. Install<a class="headerlink" href="#install" title="Permalink to this headline">¶</a></h2>
<p>If all prerequisites are satisfied then you can download the script package from <a class="reference external" href="https://github.com/MarcosFernandez/RNASeqJip">https://github.com/MarcosFernandez/RNASeqJip</a></p>
</div>
<div class="section" id="file-name-pattern">
<h2>2. File name pattern<a class="headerlink" href="#file-name-pattern" title="Permalink to this headline">¶</a></h2>
<p>RnaSeq JIP Pipeline runs <a class="reference external" href="http://lomereiter.github.io/sambamba/">markdup</a> from sambamba to calculate Duplicates statistics. Statistics are calculated per sample and library. All bam alignment files comming from the same library are previously merged to later
perfom <a class="reference external" href="http://lomereiter.github.io/sambamba/">markdup</a> and get the stats.</p>
<p>RnaSeq JIP Pipeline clusters input bam files according to the sample and library name. Then, Input bam alignment files must follow the next pattern:</p>
<blockquote>
<div><strong>Sample_Library_Flowcell_Lane[_Index_].bam</strong></div></blockquote>
<p>The output file generated follows next name pattern:</p>
<blockquote>
<div><strong>SampleName.Library.rmdup.sample.libs.flagstat</strong></div></blockquote>
<div class="admonition hint">
<p class="first admonition-title">Hint</p>
<p>If you are using <tt class="docutils literal"><span class="pre">lims</span></tt> from CNAG cluster it is recommended to run next command in order to create soft links to fastq files belonging the right name.</p>
<div class="last highlight-python"><pre>lims -s PROJECT_NAME  --status pass --link '${barcode}_${library}_${flowcell}_${lane}_${index}_${pair}.fastq.gz'</pre>
</div>
</div>
</div>
<div class="section" id="create-configuration-file">
<h2>3. Create configuration file<a class="headerlink" href="#create-configuration-file" title="Permalink to this headline">¶</a></h2>
<p>Create a configuration file with general parameters.</p>
<div class="highlight-python"><pre>gemtools rna-pipeline -q 33 -a annotation.gtf -r annotation.gtf.junctions.gem -i genome.gem -o /directory_output/  --save config.gt</pre>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p>Index Reference and annotation.</p>
<p><tt class="docutils literal"><span class="pre">genome.gem</span></tt>:  Indexed reference for the gem mapper.</p>
<p><tt class="docutils literal"><span class="pre">annotation.gtf.junctions.gem</span></tt>: Indexed annotation for the gem mapper.</p>
<p class="last">To perform the indexing of the assembly reference and its annotation you should follow steps 2 and 3 from the <a class="reference external" href="http://gemtools.github.io/docs/rna_pipeline.html">RNA Pipeline Quickstart</a>.</p>
</div>
</div>
<div class="section" id="run-rnaseq-jip-pipeline">
<h2>4. Run RnaSeq JIP Pipeline<a class="headerlink" href="#run-rnaseq-jip-pipeline" title="Permalink to this headline">¶</a></h2>
<p>Manage all the pipeline steps. gemtools mapping, flux-capacitor quantification and PCR Duplicates calculation.</p>
<div class="highlight-python"><pre>rnaseq_jip.jip -i /dir_fastq/*_1.fastq.gz -c config.gt  -a annotation.gtf -o /directory_mappings/ -q /directory_quantification/ -t 8 -- submit

--noFlux    Do not run flux-capacitor quantification.
--noPCR     Do not run PCR Duplicates.
-l          File to store the mapping directory to upload the data to a lims system.
--noFilter  Do not upload filtered bam stats to lims system.</pre>
</div>
</div>
<div class="section" id="uploading-rnaseq-results-to-lims-cnag-system">
<h2>5. Uploading RNAseq Results to Lims CNAG system<a class="headerlink" href="#uploading-rnaseq-results-to-lims-cnag-system" title="Permalink to this headline">¶</a></h2>
<div class="section" id="uploading-through-lims-daemon">
<h3>5.1. Uploading through Lims Daemon<a class="headerlink" href="#uploading-through-lims-daemon" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><p>Parameter <tt class="docutils literal"><span class="pre">-l</span></tt> from <tt class="docutils literal"><span class="pre">rnaseq_jip.jip</span></tt> allows you to configure a temporary file to add the mapping path. The goal is to notify to a <a class="reference external" href="http://statgen.cnag.cat/daemon_rna_lims/">lims daemon system</a> that there is data waiting to be upload to lims database.</p>
<p>You should select the same temporary file digested by the <a class="reference external" href="http://statgen.cnag.cat/daemon_rna_lims/">lims daemon system</a>.</p>
</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>To upload data in Lims CNAG system you need a user and a key. This user and key should be in a file called <strong>config</strong> and has to be located at the same directory where the daemon rna scripts are located.
The format of this file has to be:</p>
<p class="last"><tt class="docutils literal"><span class="pre">username</span> <span class="pre">key</span></tt></p>
</div>
</div>
<div class="section" id="uploading-through-limsrnaseq-py">
<h3>5.2. Uploading through limsRnaSeq.py<a class="headerlink" href="#uploading-through-limsrnaseq-py" title="Permalink to this headline">¶</a></h3>
<blockquote>
<div><p>You can upload RNA seq data to the lims database without the lims daemon system using <tt class="docutils literal"><span class="pre">limsRnaSeq.py</span></tt>.</p>
<blockquote>
<div><p>Just run it as:</p>
<blockquote>
<div><p><tt class="docutils literal"><span class="pre">limsRnaSeq.py</span> <span class="pre">-d</span> <span class="pre">/mapping_path/</span></tt></p>
<p>or <tt class="docutils literal"><span class="pre">limsRnaSeq.py</span> <span class="pre">-d</span> <span class="pre">/mapping_path/</span> <span class="pre">--no_filter</span></tt>    in case you do not want to use bam filtered files.</p>
</div></blockquote>
</div></blockquote>
</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>To upload data in Lims CNAG system you need a user and a key. This user and key should be in a file called <strong>config</strong> and has to be located at the same directory where the <tt class="docutils literal"><span class="pre">limsRnaSeq.py</span></tt> script is located.
The format of this file has to be:</p>
<p class="last"><tt class="docutils literal"><span class="pre">username</span> <span class="pre">key</span></tt></p>
</div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="#">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">RnaSeq JIP Pipeline</a><ul>
<li><a class="reference internal" href="#prerequisites">0. Prerequisites</a><ul>
<li><a class="reference internal" href="#download-and-install-jip-pipeline-system">0.1. Download and install JIP PIPELINE system</a></li>
<li><a class="reference internal" href="#download-and-install-gemtools-rnaseq">0.2. Download and install gemtools rnaseq</a></li>
<li><a class="reference internal" href="#flux-capacitor">0.3 Flux-Capacitor</a></li>
<li><a class="reference internal" href="#download-and-install-sambamba">0.4. Download and install SAMBAMBA</a></li>
<li><a class="reference internal" href="#lims-updating-daemon">0.5. Lims updating daemon</a></li>
</ul>
</li>
<li><a class="reference internal" href="#install">1. Install</a></li>
<li><a class="reference internal" href="#file-name-pattern">2. File name pattern</a></li>
<li><a class="reference internal" href="#create-configuration-file">3. Create configuration file</a></li>
<li><a class="reference internal" href="#run-rnaseq-jip-pipeline">4. Run RnaSeq JIP Pipeline</a></li>
<li><a class="reference internal" href="#uploading-rnaseq-results-to-lims-cnag-system">5. Uploading RNAseq Results to Lims CNAG system</a><ul>
<li><a class="reference internal" href="#uploading-through-lims-daemon">5.1. Uploading through Lims Daemon</a></li>
<li><a class="reference internal" href="#uploading-through-limsrnaseq-py">5.2. Uploading through limsRnaSeq.py</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/index.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li><a href="#">RNA Seq JIP Pipeline 0.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2015, Marcos Fernandez-Callejo.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>