.. RNA Seq JIP Pipeline documentation master file, created by
   sphinx-quickstart on Tue Jul 21 15:39:15 2015.
   You can adapt this file completely to your liking, but it should at least
   contain the root `toctree` directive.

RnaSeq JIP Pipeline
===================

Collection of `JIP pipeline system`_ scripts to manage the RNA Seq Pipeline using `gemtools rnaseq`_ for mapping, `flux`_ for quantification and `sambamba`_ to calculate PCR Duplicates statistics. It also updates a temporary file to be processed by `daemon lims`_ in order to update the pipeline results to the Lims System.

.. _JIP pipeline system: https://pyjip.readthedocs.org/en/latest/

.. _gemtools rnaseq: http://gemtools.github.io/docs/index.html

.. _flux: http://sammeth.net/confluence/display/FLUX/Home

.. _sambamba: http://lomereiter.github.io/sambamba/

.. _daemon lims: http://statgen.cnag.cat/daemon_rna_lims/



0. Prerequisites
----------------

0.1. Download and install JIP PIPELINE system
``````````````````````````````````````````````
The JIP pipeline system is a python library and a set of command line utilities that allows you to create batch-process based computational pipeline that can be submitted and managed on a computer cluster or on your local machine.

Install the jip pipeline system following the instructions found in: `http://pyjip.readthedocs.org/en/latest/`_


0.2. Download and install gemtools rnaseq 
`````````````````````````````````````````

Get gemtools rnaseq from `http://gemtools.github.io/`_ and follow install instructions.

.. _http://gemtools.github.io/: http://gemtools.github.io/


0.3 Flux-Capacitor
```````````````````
Considers the read abundance on an exonic segment as the cumulative abundance of all transcript isoforms. The transcript is represented as a combination of exons and aims at estimating individual transcript abundance from the observed read counts at each exon.

Download and Install flux capacitor from `http://sammeth.net/confluence/display/FLUX/2+-+Download`_

.. _http://sammeth.net/confluence/display/FLUX/2+-+Download: http://sammeth.net/confluence/display/FLUX/2+-+Download


0.4. Download and install SAMBAMBA
```````````````````````````````````

Sambamba is a tool for processing SAM/BAM files in a fast way. It is used for merging, calculate PCR Duplicates and general statistics from BAM files.

Download and install the package in your system from: `https://github.com/lomereiter/sambamba/releases`_


.. _http://pyjip.readthedocs.org/en/latest/: http://pyjip.readthedocs.org/en/latest/

.. _https://github.com/lomereiter/sambamba/releases: https://github.com/lomereiter/sambamba/releases


0.5. Lims updating daemon
`````````````````````````
If you want to use the lims updating system to upload the data generated by the pipeline you must install and start the `rna updating lims daemon`_.

.. _rna updating lims daemon: http://statgen.cnag.cat/daemon_rna_lims/


1. Install
----------

If all prerequisites are satisfied then you can download the script package from `https://github.com/MarcosFernandez/RNASeqJip`_

.. _https://github.com/MarcosFernandez/RNASeqJip: https://github.com/MarcosFernandez/RNASeqJip
 


2. File name pattern
--------------------

RnaSeq JIP Pipeline runs `markdup`_ from sambamba to calculate Duplicates statistics. Statistics are calculated per sample and library. All bam alignment files comming from the same library are previously merged to later 
perfom `markdup`_ and get the stats.

RnaSeq JIP Pipeline clusters input bam files according to the sample and library name. Then, Input bam alignment files must follow the next pattern:

    **Sample_Library_Flowcell_Lane[_Index_].bam**

The output file generated follows next name pattern:

    **SampleName.Library.rmdup.sample.libs.flagstat**

.. hint::

    If you are using ``lims`` from CNAG cluster it is recommended to run next command in order to create soft links to fastq files belonging the right name. ::

        lims -s PROJECT_NAME  --status pass --link '${barcode}_${library}_${flowcell}_${lane}_${index}_${pair}.fastq.gz'

.. _markdup: http://lomereiter.github.io/sambamba/


3. Create configuration file
----------------------------

Create a configuration file with general parameters.

::

    gemtools rna-pipeline -q 33 -a annotation.gtf -r annotation.gtf.junctions.gem -i genome.gem -o /directory_output/  --save config.gt

.. warning::
    
    Index Reference and annotation.

    ``genome.gem``:  Indexed reference for the gem mapper.

    ``annotation.gtf.junctions.gem``: Indexed annotation for the gem mapper. 

    To perform the indexing of the assembly reference and its annotation you should follow steps 2 and 3 from the `RNA Pipeline Quickstart`_.


.. _RNA Pipeline Quickstart: http://gemtools.github.io/docs/rna_pipeline.html


4. Run RnaSeq JIP Pipeline
--------------------------

Manage all the pipeline steps. gemtools mapping, flux-capacitor quantification and PCR Duplicates calculation.

:: 

    rnaseq_jip.jip -i /dir_fastq/*_1.fastq.gz -c config.gt  -a annotation.gtf -o /directory_mappings/ -q /directory_quantification/ -t 8 -- submit 

    --noFlux    Do not run flux-capacitor quantification.
    --noPCR     Do not run PCR Duplicates.
    -l          File to store the mapping directory to upload the data to a lims system

.. note::

    **Uploading RNAseq Results to Lims CNAG system.**
                                            

    **1. Uploading through Lims Daemon** 

    ``-l`` Allows you to configure a temporary file to add the mapping path. The goal is to notify to a `lims daemon system`_ that there is data waiting to be upload to lims database.
       
    You should select the same temporary file digested by the `lims daemon system`_.

    **2. Uploading through limsRnaSeq.py**

    You can upload RNA seq data to the lims database without the lims daemon system using ``limsRnaSeq.py``.

        Just run it as:

            ``limsRnaSeq.py -d /mapping_path/``

            or ``limsRnaSeq.py -d /mapping_path/ --no_filter``    in case you do not want to use bam filtered files.
    

.. _lims daemon system: http://statgen.cnag.cat/daemon_rna_lims/
